board.py

```diff
@@ -16,38 +17,118 @@ def __ne__(self, other):
 
     def __hash__(self):
         return hash(self.name)
+    
+    def as_dict(self):
+        return {"name":   str(self.name),
+                "owner":  str(self.owner),
+                "troops": str(self.troops)}
 
+    def to_json(self):
+        return dumps(self.as_dict())
 
+    def from_dict(dict_rep, name_keys=["name"], owner_keys=["owner"],
+            troop_keys=["troops"]):
+        name = None
+        owner = None
+        troops = 0
+        for name_key in name_keys:
+            if name_key in dict_rep.keys():
+                name = dict_rep[name_key]
+                break
+        for owner_key in owner_keys:
+            if owner_key in dict_rep.keys():
+                owner = dict_rep[owner_key]
+                break
+        for troop_key in troop_keys:
+            if troop_key in dict_rep.keys():
+                troops = dict_rep[troop_key]
+                break
+        if name:
+            return Country(name, owner, troops)
+        else:
+            # TODO: raise an exception?
+            return None
+
+    def from_json(json_str):
+        return Country.from_dict(loads(json_str))
+        
 class Board(object):
-    """A Board of the Risk game. Manages ownership of countries etc."""
+    # A Board of the Risk game. Manages ownership of countries etc.
     def __init__(self):
-        # TODO: builder or parser for Maps
-
-        # small static Map
-        r"""
-        ger -- usa
-         |  \   |
-        tha    aus
-        """
-
-        ger = Country('Germany')
-        usa = Country('USA')
-        tha = Country('Thailand')
-        aus = Country('Australia')
-
-        self.country_names = {
-            'Germany': ger,
-            'USA': usa,
-            'Thailand': tha,
-            'Australia': aus
-        }
-        self.countries = [ger, usa, tha, aus]
-        self.edges = defaultdict(list, {
-            ger: [usa, aus, tha],
-            usa: [ger, aus],
-            tha: [ger],
-            aus: [ger, usa]
-        })
+        # countries: dict name -> country
+        self.countries = {}
+        # continents: dict name -> set of countries
+        self.continents = {}
+        # edges: set (name1, name2)
+        self.edges = set()
+
+    def __eq__(self, other):
+        return self.countries == other.countries and self.edges == other.edges
+
+    def add_country(self, country_name, continent_name, owner=None, troops=0):
```

Comment:
I'd prefer a board's countries to be immutable after instantiation because a map never won't change during a game.